addons/mail/static/src/core/common/message.js

```diff
@@ -437,7 +437,17 @@ export class Message extends Component {
     prepareMessageBody(element) {}
 
     enterEditMode() {
+        const body = document.createElement("span");
+        body.innerHTML = this.props.message.body;
         const messageContent = convertBrToLineBreak(this.props.message.body);
+        const mentionedChannelElements = body.querySelectorAll(".o_channel_redirect");
+        this.props.message.mentionedChannelPromises = Array.from(mentionedChannelElements).map(
+            async (el) =>
+                this.store.Thread.getOrFetch({
+                    id: el.dataset.oeId,
+                    model: el.dataset.oeModel,
+                })
+        );
```

Comment:
This should only be done if `oeModel` is `discuss.channel`, and this whole diff should be in `discuss` folder.

Reply:
in `getOrFetch`, the model has already been checked so there is no need to check this again here?

Reply:
The base method works for any thread, and it would allow an existing thread to pass-through to channel mentions even though it's not a channel.